// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTContractEnums";
option optimize_for = LITE_RUNTIME;

enum AddressType {
    ADDRESSTYPE_ERROR = 0;
    ADDRESSTYPE_IPV4 = 1;
    ADDRESSTYPE_IPV6 = 2;
    ADDRESSTYPE_ONION = 3;
    ADDRESSTYPE_EEP = 4;
    ADDRESSTYPE_INPROC = 5;
}

enum ProtocolVersion {
    PROTOCOLVERSION_ERROR = 0;
    PROTOCOLVERSION_LEGACY = 1;
    PROTOCOLVERSION_NOTIFY = 2;
}

enum UnitType {
    UNITTYPE_ERROR = 0;
    UNITTYPE_CURRENCY = 1;
    UNITTYPE_SECURITY = 2;
    UNITTYPE_BASKET = 3;
}

enum EquityType {
    EQUITYTYPE_ERROR = 0;
    EQUITYTYPE_SHARES = 1;
}
